src/Servers/Kestrel/Core/src/Internal/WebTransport/WebTransportStream.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.IO.Pipelines;
+using System.Net.Http;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Core.WebTransport;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.WebTransport;
+
+internal sealed class WebTransportStream : ConnectionContext, IStreamDirectionFeature
+{
+    private readonly CancellationTokenRegistration _connectionClosedRegistration;
+    private readonly bool _canWrite;
+    private readonly bool _canRead;
+    private readonly DuplexPipe _duplexPipe;
+    private readonly IFeatureCollection _features;
+    private readonly KestrelTrace _log;
+    private IDictionary<object, object?>? _items;
+    private bool _isClosed;
+
+    public readonly long StreamId;
+
+    public override string ConnectionId { get => StreamId.ToString(NumberFormatInfo.InvariantInfo); set => throw new NotSupportedException(); }
+
+    public override IDuplexPipe Transport { get => _duplexPipe; set => throw new NotSupportedException(); }
+
+    public override IFeatureCollection Features => _features;
+
+    public override IDictionary<object, object?> Items
+    {
+        get => _items ??= new Dictionary<object, object?>();
+        set => _items = value;
+    }
```

Comment:
I think this should be reading IConnectionItemsFeature from the feature collection.

Reply:
Hmm, yeah ok. I just looked at the `DefaultConnectionContext` class and seems I can just do what they do. Basically, the class becomes the feature.